client/my-sites/site-settings/header-code/index.jsx

```diff
@@ -0,0 +1,182 @@
+import EmptyContent from 'calypso/components/empty-content';
+import HeaderCake from 'calypso/components/header-cake';
+import InlineSupportLink from 'calypso/components/inline-support-link';
+import Main from 'calypso/components/main';
+import NavigationHeader from 'calypso/components/navigation-header';
+import { successNotice, errorNotice } from 'calypso/state/notices/actions';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import getSiteSetting from 'calypso/state/selectors/get-site-setting';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
+import { saveSiteSettings } from 'calypso/state/site-settings/actions';
+import {
+	isRequestingSiteSettings,
+	isSavingSiteSettings,
+	isSiteSettingsSaveSuccessful,
+	getSiteSettingsSaveError,
+} from 'calypso/state/site-settings/selectors';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import './style.scss';
+
+const HeaderCodeSettings = ( {
+	errorMessage,
+	hasPlanFeature,
+	headCode,
+	isAdmin,
+	isJetpack,
+	isRequestingSettings,
+	isSaveFailure,
+	isSaveSuccess,
+	isSavingSettings,
+	saveSettings,
+	siteId,
+	siteSlug,
+	successMessage,
+	translate,
+} ) => {
+	const [ htmlCode, setHtmlCode ] = useState( '' );
+
+	useEffect( () => {
+		if ( 'string' === typeof headCode && headCode.length ) {
+			setHtmlCode( headCode );
+		}
+
+		// Necessary for if switching to a site without any code.
+		if ( ! headCode ) {
+			setHtmlCode( '' );
+		}
+	}, [ headCode, siteId ] );
+
+	useEffect( () => {
+		if ( isSaveSuccess ) {
+			successMessage( translate( 'Code updated successfully!' ), {
+				duration: 4000,
+			} );
+		} else if ( isSaveFailure && ! isSavingSettings ) {
+			errorMessage( translate( 'There was an error saving your code.' ), {
+				duration: 4000,
+			} );
+		}
+	}, [ isSaveSuccess, isSavingSettings, isSaveFailure ] );
+
+	const handleSaveSettings = () => {
+		saveSettings( siteId, {
+			jetpack_header_code: htmlCode,
+		} );
+	};
+
+	const isSupported = isJetpack || hasPlanFeature;
+
+	return (
+		<Main className="header-code">
+			<DocumentHead title={ translate( 'Header Code' ) } />
+			{ siteId && <QuerySiteSettings siteId={ siteId } /> }
+			{ ! isAdmin && (
+				<EmptyContent
+					illustration={ illustration404 }
+					title={ translate( 'You are not authorized to view this page' ) }
+				/>
+			) }
+
+			{ isAdmin && (
+				<>
+					<NavigationHeader
+						title={ translate( 'Header Code' ) }
+						subtitle={ translate( 'Insert code within the Header element of your site.' ) }
+					/>
+					<HeaderCake backHref={ `/settings/general/${ siteSlug }` } isCompact>
+						<h1>{ translate( 'Header Code' ) }</h1>
+					</HeaderCake>
+					<Card>
+						<UpsellNudge
+							plan={ PLAN_BUSINESS }
+							title={ translate( 'Add custom code to your site with the %(planName)s plan.', {
```

Comment:
Is this really the case? Do we really need a business plan to insert the header code?

Reply:
Yes, it'll be stripped out on WP.com Simple sites.

Reply:
What about Personal (AKA Starter) and Premium (AKA Explorer) plans? Does only the WP.com business plan have this feature?

Reply:
Yes, indeed. I believe that's for security reasons since Simple sites share the platform alongside product reasons (but of course, that's been determined well above me!). 

Users wouldn't even be able to insert such code with the Custom HTML block: https://wordpress.com/support/wordpress-editor/blocks/custom-html-block/#supported-html-tags